src/main/java/seedu/address/model/person/Gender.java

```diff
@@ -0,0 +1,61 @@
+package seedu.address.model.person;
+
+import static java.util.Objects.requireNonNull;
+
+import seedu.address.commons.exceptions.IllegalValueException;
+
+/**
+ * Represents a Person's gender in the address book.
+ * Guarantees: immutable; is valid as declared in {@link #isValidGender(String)}
+ */
+public class Gender {
+
+    public static final String MESSAGE_GENDER_CONSTRAINTS =
+            "Person gender should be a string of either 'Male', 'Female' or 'M', 'F'";
+    public static final String GENDER_VALIDATION_WORD1 = "Male";
+    public static final String GENDER_VALIDATION_WORD2 = "Female";
+    public static final String GENDER_VALIDATION_WORD3 = "M";
+    public static final String GENDER_VALIDATION_WORD4 = "F";
+
+    public final String value;
+
+    /**
+     * Validates given gender.
+     *
+     * @throws IllegalValueException if given gender string is invalid.
+     */
+    public Gender(String gender) throws IllegalValueException {
+        requireNonNull(gender);
+        String trimmedGender = gender.trim();
+        if (!isValidGender(trimmedGender)) {
+            throw new IllegalValueException(MESSAGE_GENDER_CONSTRAINTS);
+        }
+        this.value = trimmedGender;
+    }
+
+    /**
+     * Returns if a given string is a valid person gender.
+     */
+    public static boolean isValidGender(String test) {
+        return test.equals(GENDER_VALIDATION_WORD1) || test.equals(GENDER_VALIDATION_WORD2)
+                || test.equals(GENDER_VALIDATION_WORD3) || test.equals(GENDER_VALIDATION_WORD4);
+    }
+
+    @Override
+    public String toString() {
+        return value;
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        return other == this // short circuit if same object
+                || (other instanceof Gender // instanceof handles nulls
+                && this.value.equals(((Gender) other).value)); // state check
```

Comment:
for this, should we be checking if Male = M and Female = F?

Reply:
haha, I'll convert M into 'Male', and F into 'Female', then there is no problem here!! thanks a lot